### PR TITLE
Add clinician new modal

### PR DESCRIPTION
**Pull request by paulfalgout**
Also change role from a named attribute to a relationship

This turned out to be quite a bit more work than initially expected

Shortcut Story ID: [sc-31295]

### CODE REVIEW

File: src/js/views/clinicians/clinician-modal/clinician-modal_views.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import { extend } from 'underscore';
```
Comment (paulfalgout) [inline src/js/views/clinicians/clinician-modal/clinician-modal_views.js:1]: FWIW this is very very similar to the add patient modal views

File: src/js/views/clinicians/shared/clinicians_views.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ const i18n = intl.clinicians.shared.clinicianViews;
 
 const GroupsComponent = GroupsManagerComponent.extend({
   removeMemberGroup(group) {
+    if (this.member.isNew()) {
```
Comment (paulfalgout) [inline src/js/views/clinicians/shared/clinicians_views.js:19]: Previously this component wasn't available to a new clinician, but we don't want the warning below in that situation

File: src/js/views/globals/root_views.js [hunk 3]
```diff
@@ -306,10 +306,10 @@ const RootView = CollectionView.extend({
     // Add lowest layer (z-index) to highest
     this.addChildView(this.appView);
     this.addRegionView('tooltip', new TooltipRegionView({ $body }));
-    this.addRegionView('alert', new TopRegionView());
     this.addRegionView('modalSidebar', new ModalSidebarRegionView({ $body }));
     this.addRegionView('modal', new ModalRegionView({ $body }));
     this.addRegionView('modalSmall', new ModalRegionView({ $body }));
+    this.addRegionView('alert', new TopRegionView());
```
Comment (paulfalgout) [inline src/js/views/globals/root_views.js:312]: alerts were rendering below the modal and darkening.. I can't think of a reason why that should be true.

File: test/integration/clinicians/clinician-modal.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    name: 'Group One',
+  },
+  {
+    id: '2',
+    name: 'Group Two',
+  },
+];
+
+context('clinicians modal', function() {
+  specify('add clinician', function() {
```
Comment (paulfalgout) [inline test/integration/clinicians/clinician-modal.js:19]: this was adapted from the similar tests when we were adding them from the sidebar

**Review by nmajor25 — APPROVED** [review]

**Pull request merged**